README.rst
Python Client for Apigee Connect API API
========================================

|stable| |pypi| |versions|

`Apigee Connect API API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-apigee-connect.svg
   :target: https://pypi.org/project/google-cloud-apigee-connect/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-apigee-connect.svg
   :target: https://pypi.org/project/google-cloud-apigee-connect/
.. _Apigee Connect API API: https://cloud.google.com/apigee/docs/hybrid/v1.4/apigee-connect
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/apigeeconnect/latest
.. _Product Documentation:  https://cloud.google.com/apigee/docs/hybrid/v1.4/apigee-connect

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Apigee Connect API API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Apigee Connect API API.:  https://cloud.google.com/apigee/docs/hybrid/v1.4/apigee-connect
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-apigee-connect


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-apigee-connect

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Apigee Connect API API
   to see other available methods on the client.
-  Read the `Apigee Connect API API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Apigee Connect API API Product documentation:  https://cloud.google.com/apigee/docs/hybrid/v1.4/apigee-connect
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [1.0.1](https://www.github.com/googleapis/python-apigee-connect/compare/v1.0.0...v1.0.1) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([50db178](https://www.github.com/googleapis/python-apigee-connect/commit/50db178c9dec3b1214d537da0a9bd9088d9ebf43))


### Documentation

* migrate default branch from master to main ([#33](https://www.github.com/googleapis/python-apigee-connect/issues/33)) ([1d89407](https://www.github.com/googleapis/python-apigee-connect/commit/1d894071b496e7999b96f5f18e65499857f18412))

## [1.0.0](https://www.github.com/googleapis/python-apigee-connect/compare/v0.2.2...v1.0.0) (2021-08-03)


### Features

* bump release level to production/stable ([#28](https://www.github.com/googleapis/python-apigee-connect/issues/28)) ([65803db](https://www.github.com/googleapis/python-apigee-connect/commit/65803db4821b594124bfb88012ba4f954568a895))

### [0.2.2](https://www.github.com/googleapis/python-apigee-connect/compare/v0.2.1...v0.2.2) (2021-07-29)


### Bug Fixes

* enable self signed jwt for grpc ([#24](https://www.github.com/googleapis/python-apigee-connect/issues/24)) ([d02ffa1](https://www.github.com/googleapis/python-apigee-connect/commit/d02ffa1f4906da6ba8d479b18947e6d28ccb1987))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#20](https://www.github.com/googleapis/python-apigee-connect/issues/20)) ([6f37510](https://www.github.com/googleapis/python-apigee-connect/commit/6f37510de8b5bd00720335f1343354ee2c932d9e))


### Miscellaneous Chores

* release as 0.2.2 ([#25](https://www.github.com/googleapis/python-apigee-connect/issues/25)) ([f6660d3](https://www.github.com/googleapis/python-apigee-connect/commit/f6660d3d604c61b748abe59a51923b74b687d761))

### [0.2.1](https://www.github.com/googleapis/python-apigee-connect/compare/v0.2.0...v0.2.1) (2021-07-21)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#19](https://www.github.com/googleapis/python-apigee-connect/issues/19)) ([cfa46e7](https://www.github.com/googleapis/python-apigee-connect/commit/cfa46e7528595d57f396839ef167cb8ede29981b))

## [0.2.0](https://www.github.com/googleapis/python-apigee-connect/compare/v0.1.0...v0.2.0) (2021-07-14)


### Features

* add always_use_jwt_access ([#11](https://www.github.com/googleapis/python-apigee-connect/issues/11)) ([bdd0f21](https://www.github.com/googleapis/python-apigee-connect/commit/bdd0f2109e34d100c285ab355f302326816f24c8))


### Bug Fixes

* disable always_use_jwt_access ([f4cff83](https://www.github.com/googleapis/python-apigee-connect/commit/f4cff83baf1865477139bd14b02a12e47505150d))
* disable always_use_jwt_access ([#15](https://www.github.com/googleapis/python-apigee-connect/issues/15)) ([f4cff83](https://www.github.com/googleapis/python-apigee-connect/commit/f4cff83baf1865477139bd14b02a12e47505150d))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-apigee-connect/issues/1127)) ([#6](https://www.github.com/googleapis/python-apigee-connect/issues/6)) ([a465d9a](https://www.github.com/googleapis/python-apigee-connect/commit/a465d9a37f4738f0f16e89b4c26a74366c0834fb)), closes [#1126](https://www.github.com/googleapis/python-apigee-connect/issues/1126)

## 0.1.0 (2021-06-13)


### Features

* generate v1 ([b52b2bf](https://www.github.com/googleapis/python-apigee-connect/commit/b52b2bf1f8456eb7fc815857d90384d2f596d23a))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/apigeeconnect_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
